deep_qa/data/instances/sentence_selection_instance.py

```diff
@@ -0,0 +1,161 @@
+from typing import Dict, List
+
+import numpy as np
+from overrides import overrides
+
+from .instance import TextInstance, IndexedInstance
+from ..data_indexer import DataIndexer
+
+
+class SentenceSelectionInstance(TextInstance):
+    """
+    A SentenceSelectionInstance an instance for the sentence selection
+    task. A SentenceSelectionInstance stores a question as a string, and a set of sentences
+    as a list of strings. The labels is a single int, indicating the index of
+    the sentence that contains the answer to the question.
+    """
+    def __init__(self, question_text: str, sentences: List[str], label: int, index: int=None):
+        super(SentenceSelectionInstance, self).__init__(label, index)
+        self.question_text = question_text
+        self.sentences = sentences
+
+    def __str__(self):
+        return ('SentenceSelectionInstance(' + self.question_text +
+                ', ' + self.sentences + ', ' +
+                str(self.label) + ')')
+
+    @overrides
+    def words(self) -> Dict[str, List[str]]:
+        words = self._words_from_text(self.question_text)
+        sentences_words = [self._words_from_text(sentence) for sentence in
+                           self.sentences]
+        for namespace in words:
+            for sentence_words in sentences_words:
+                words[namespace].extend(sentence_words[namespace])
+        return words
+
+    @overrides
+    def to_indexed_instance(self, data_indexer: DataIndexer):
+        question_indices = self._index_text(self.question_text, data_indexer)
+        sentences_indices = [self._index_text(sentence, data_indexer) for
+                             sentence in self.sentences]
+        return IndexedSentenceSelectionInstance(question_indices,
+                                                sentences_indices,
+                                                self.label,
+                                                self.index)
+
+    @classmethod
+    def read_from_line(cls, line: str, default_label: bool=None):
+        """
+        Reads a SentenceSelectionInstance object from a line.
+        The format has one of two options:
+
+        (1) [example index][tab][question][tab][list_of_sentences][tab][label]
+        (2) [question][tab][list_of_sentences][tab][label]
+
+        The ``list_of_sentences`` column is assumed formatted as: ``[sentence]###[sentence]###[sentence]...``
+        That is, we split on three hashes (``"###"``).
+
+        ``default_label`` is ignored, but we keep the argument to match the interface.
```

Comment:
We should just remove `default_label` from the interface, actually.  I removed the ability to _use_ `default_label` from the trainer code a long time ago.  But you shouldn't worry about that here; I'll open an issue for it.